lib/board.rb

```diff
@@ -0,0 +1,43 @@
+class Board
+  attr_accessor :cells
+
+  def initialize()
+    reset!
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def full?
+    cells.all?{|token| token == "X" || token == "O"}

```

Comment:
Nice job using the `all?` iterator!! This can be made even more efficient. `Board#full?` can be read as checking whether all elements are not empty strings:

`cells.all? {|cell| cell != " "}`